libs/vkd3d/state.c

```diff
@@ -3466,12 +3482,15 @@ HRESULT vkd3d_bindless_state_init(struct vkd3d_bindless_state *bindless_state,
         goto fail;
     }
 
+    if (bindless_state->flags & VKD3D_RAW_VA_UAV_COUNTER)
+        extra_bindings |= VKD3D_BINDLESS_SET_EXTRA_UAV_COUNTER_BUFFER;
+
     if (FAILED(hr = vkd3d_bindless_state_add_binding(bindless_state, device,
             VKD3D_BINDLESS_SET_SAMPLER, VK_DESCRIPTOR_TYPE_SAMPLER)))
         goto fail;
 
     if (FAILED(hr = vkd3d_bindless_state_add_binding(bindless_state, device,
-            VKD3D_BINDLESS_SET_CBV,
+            VKD3D_BINDLESS_SET_CBV | extra_bindings,
```

Comment:
Why here and not VKD3D_BINDLESS_SET_UAV | BUFFER? Ideally we shouldn't have to bind the CBV set only for the sake of using UAVs.

Reply:
Putting it into `VKD3D_BINDLESS_SET_UAV | BUFFER` actually seems worse if the SSBO path is enabled since we'd always bind the texel buffer set, which is less likely to be used in a shader that uses UAV counters than the CBV set.

Of course we can conditionally add it either to the `UAV | BUFFER` or `UAV | RAW_SSBO` set depending on whether the SSBO path is enabled, but that's a lot of extra logic just to decide where to put the binding and *potentially* save one SGPR, not really sure if that's really worth it.

Reply:
Ok.